src/allocs_model.jl

```diff
@@ -0,0 +1,153 @@
+export test_allocs_nlpmodels, print_nlp_allocations
+
+"""
+    test_allocs_nlpmodels(nlp::AbstractNLPModel; exclude = [])
+
+Returns a `Dict` containing allocations of the in-place functions of NLPModel API.
+
+The keyword `exclude` takes a Array of Function to be excluded from the tests. Use `hess` (resp. `jac`) to exclude `hess_coord` and `hess_structure` (resp. `jac_coord` and `jac_structure`).
+"""
+function test_allocs_nlpmodels(nlp::AbstractNLPModel; exclude = [])
+  nlp_allocations = Dict(
+    :obj => NaN,
+    :grad! => NaN,
+    :hess_structure! => NaN,
+    :hess_coord! => NaN,
+    :hprod! => NaN,
+    :hess_op! => NaN,
+    :hess_op_prod! => NaN,
+    :cons! => NaN,
+    :jac_structure! => NaN,
+    :jac_coord! => NaN,
+    :jac_op! => NaN,
+    :jac_op_prod! => NaN,
+    :jac_op_transpose_prod! => NaN,
+    :hess_lag_coord! => NaN,
+    :hprod_lag! => NaN,
+    :hess_lag_op! => NaN,
+    :hess_lag_op_prod! => NaN,
+  )
+
+  if !(obj in exclude)
+    x = get_x0(nlp)
+    obj(nlp, x)
+    nlp_allocations[:obj] = @allocated obj(nlp, x)
+  end
+  if !(grad in exclude)
+    x = get_x0(nlp)
+    g = similar(x)
+    grad!(nlp, x, g)
+    nlp_allocations[:grad!] = @allocated grad!(nlp, x, g)
+  end
+  if !(hess in exclude)
+    rows = Vector{Int}(undef, nlp.meta.nnzh)
+    cols = Vector{Int}(undef, nlp.meta.nnzh)
+    hess_structure!(nlp, rows, cols)
+    nlp_allocations[:hess_structure!] = @allocated hess_structure!(nlp, rows, cols)
+    x = get_x0(nlp)
+    vals = Vector{eltype(x)}(undef, nlp.meta.nnzh)
+    hess_coord!(nlp, x, vals)
+    nlp_allocations[:hess_coord!] = @allocated hess_coord!(nlp, x, vals)
+    if get_ncon(nlp) > 0
+      y = get_y0(nlp)
+      hess_coord!(nlp, x, y, vals)
+      nlp_allocations[:hess_lag_coord!] = @allocated hess_coord!(nlp, x, y, vals)
+    end
+  end
+  if !(hprod in exclude)
+    x = get_x0(nlp)
+    v = copy(x)
+    Hv = similar(x)
+    hprod!(nlp, x, v, Hv)
+    nlp_allocations[:hprod!] = @allocated hprod!(nlp, x, v, Hv)
+    if get_ncon(nlp) > 0
+      y = get_y0(nlp)
+      hprod!(nlp, x, y, v, Hv)
+      nlp_allocations[:hprod_lag!] = @allocated hprod!(nlp, x, y, v, Hv)
+    end
+  end
+  if !(hess_op in exclude)
+    # First we test the definition of the operator
+    x = get_x0(nlp)
+    Hv = similar(x)
+    hess_op!(nlp, x, Hv)
+    nlp_allocations[:hess_op!] = @allocated hess_op!(nlp, x, Hv)
+    if get_ncon(nlp) > 0
+      y = get_y0(nlp)
+      hess_op!(nlp, x, y, Hv)
+      nlp_allocations[:hess_lag_op!] = @allocated hess_op!(nlp, x, y, Hv)
+    end
+    # Then, we test the product
+    v = copy(x)
+    H = hess_op!(nlp, x, Hv)
+    H * v
+    nlp_allocations[:hess_op_prod!] = @allocated H * v
```

Comment:
@tmigot 
You need to use `mul!` to remove allocatations.
`H * v` always allocates with LinearOperators v2.0.

Reply:
Thanks!